docs/docs/training-data-format.mdx

```diff
@@ -428,6 +428,42 @@ policies learn to predict the next action based on a *combination* of
 both the intent and entities (you can, however, change this behavior
 using the [`use_entities`](#entities) attribute).
 
+If you use `user:` key while writing stories, you do not have to deal with the specific
+intents of the messages that are extracted by the NLU pipeline. Instead, you can simply put
+the text of the user message.
+We call it end-to-end training because actual text is used for predictions.
```

Comment:
What I don't understand as a user is how this affects the story prediction. How is it  decided if a user message is close enough to the one ive provided to follow that story? Is it essentially "whatever this message would be classified as intent-wise based on my other NLU data (does this entry get added to training data as well?), if i get another message that falls in that intent, then this story is relevant?"

I guess this format only applies to TED so maybe that doesn't make sense, but I think we could use a bit of clarification here

Reply:
Also, is this key allowed in rules, and if not, will it error as expected?

Reply:
you can have it rules. then it'll be string matching. I keep the idea that Rules is exactly the same as stories

Reply:
added explanations